component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/cloud/DefaultFaultMetricEventBuilder.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.am.analytics.publisher.reporter.cloud;
+
+import org.wso2.am.analytics.publisher.exception.MetricReportingException;
+import org.wso2.am.analytics.publisher.reporter.AbstractMetricEventBuilder;
+import org.wso2.am.analytics.publisher.reporter.MetricEventBuilder;
+import org.wso2.am.analytics.publisher.reporter.MetricSchema;
+import org.wso2.am.analytics.publisher.util.Constants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Builder class for fault events
+ */
+public class DefaultFaultMetricEventBuilder extends AbstractMetricEventBuilder {
+    private final Map<String, Class> requiredAttributes;
+    private Map<String, Object> eventMap;
+
+    protected DefaultFaultMetricEventBuilder() {
+        requiredAttributes = DefaultInputValidator.getInstance().getEventProperties(MetricSchema.ERROR);
+        eventMap = new HashMap<>();
+    }
+
+    @Override
+    public boolean validate() throws MetricReportingException {
+        for (Map.Entry<String, Class> entry : requiredAttributes.entrySet()) {
+            Object attribute = eventMap.get(entry.getKey());
+            if (attribute == null) {
+                throw new MetricReportingException(entry.getKey() + " is missing in metric data. This metric event "
+                                                           + "will not be processed further.");
+            } else if (!attribute.getClass().equals(entry.getValue())) {
```

Comment:
What happens when the passed value is null? In Async APIs we might come across places where we can't put responseMediation, requestMediation latencies. So some attributes might get left null.